todo_app/app.py

```diff
@@ -25,9 +48,20 @@ def index():
             items.sort(key=lambda x: x.due_date)
         elif request.values.get('sort') == '2':
             items.sort(key=lambda x: x.due_date, reverse=True)
-        return render_template("index.html",View_Model=get_view_model, todays_date=todays_date)
+
+        if hasattr('current_user', 'role'):
+            if current_user.role == 'writer':
+                return render_template("index.html",View_Model=get_view_model, todays_date=todays_date)
+            elif current_user.role == 'reader':
+                return render_template('index_read.html', View_Model=get_view_model, todays_date=todays_date)
+            else:
+                return render_template('index.html', View_Model=get_view_model, todays_date=todays_date)
+        else:
+            return render_template('index.html', View_Model=get_view_model, todays_date=todays_date)
+
 
     @app.route('/new_item', methods=['POST'])
+    @login_required
```

Comment:
We should also add some sort of role check within each of these functions (which we could do with a custom decorator or by checking directly inside each block). As they currently stand, it's very easy for readers to nefariously get write access:
* If they're accidentally served the wrong template then there's no protection
* If they know/guess the appropriate endpoint then they can bypass needing the right template
We definitely need to have role checks implemented server side for appropriate functions